### PR TITLE
LPS-169758 buildService

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-169696
Related issue: https://issues.liferay.com/browse/LPS-169758

Previous work/review: https://github.com/liferay-objects/liferay-portal/pull/1430

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java [hunk 1]
```diff
@@ -343,6 +372,28 @@ private NotificationTemplate _toNotificationTemplate(
...
+
+							if (objectDefinitionId == 0) {
+								return StringPool.BLANK;
+							}
+
+							ObjectDefinition objectDefinition =
+								_objectDefinitionLocalService.
+									fetchObjectDefinition(objectDefinitionId);
+
+							return objectDefinition.getExternalReferenceCode();
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java:393]: @marcelabc do we need to check if `objectDefinition` is `null`?

File: modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java [hunk 2]
```diff
@@ -37,18 +46,46 @@
...
+		}
+
+		if (attachmentObjectFieldIds.isEmpty()) {
+			notificationContext.setAttachmentObjectFieldIds(
+				ListUtil.fromArray(
+					notificationTemplate.getAttachmentObjectFieldIds()));
+		}
+		else {
+			notificationContext.setAttachmentObjectFieldIds(
+				attachmentObjectFieldIds);
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-rest-api/src/main/java/com/liferay/notification/rest/dto/v1_0/util/NotificationUtil.java:86]: @marcelabc I would keep the the alphabetical order, i.e., `setAttachmentObjectFieldIds` before `setType`

File: modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java [hunk 3]
```diff
@@ -343,6 +372,28 @@ private NotificationTemplate _toNotificationTemplate(
 				name = serviceBuilderNotificationTemplate.getName();
 				name_i18n = LocalizedMapUtil.getLanguageIdMap(
 					serviceBuilderNotificationTemplate.getNameMap());
+
+				if (GetterUtil.getBoolean(
+						PropsUtil.get("feature.flag.LPS-168220"))) {
+
+					setObjectDefinitionERC(
```
Comment (carolmariaabb) [inline modules/apps/notification/notification-rest-impl/src/main/java/com/liferay/notification/rest/internal/resource/v1_0/NotificationTemplateResourceImpl.java:379]: @marcelabc I think the pattern is to leave the `set` methods  at the end, like in [here](https://github.com/liferay/liferay-portal/blob/8c67c06639def90c044ce511e6b31a1a8e38f3d5/modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java#L725-L742)

**Comment by marcelabc** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Sent to Bchan: https://github.com/brianchandotcom/liferay-portal/pull/127048

**Pull request closed without merging**